src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -272,6 +201,66 @@ protected void containerIsStarting(final InspectContainerResponse containerInfo,
      * @return bootstrap servers
      */
     public String getBootstrapServers() {
-        return String.format("PLAINTEXT://%s:%s", getContainerIpAddress(), this.kafkaExposedPort);
+        return String.format("PLAINTEXT://%s:%s", getContainerIpAddress(), kafkaDynamicKafkaPort);
+    }
+
+    public StrimziKafkaContainer withKafkaConfigurationMap(final Map<String, String> kafkaConfigurationMap) {
+        this.kafkaConfigurationMap = kafkaConfigurationMap;
+        return this;
+    }
+
+    /**
+     * Fluent method, which sets @code{externalZookeeperConnect}.
+     *
+     * If the broker was created using Kraft, this method throws an {@link IllegalArgumentException}.
+     *
+     * @param externalZookeeperConnect connect string
+     * @return StrimziKafkaContainer instance
+     */
+    public StrimziKafkaContainer withExternalZookeeperConnect(final String externalZookeeperConnect) {
+        if (useKraft) {
+            throw new IllegalArgumentException("Cannot configure an external Zookeeper and use Kraft at the same time");
+        }
+        this.externalZookeeperConnect = externalZookeeperConnect;
+        return self();
+    }
+
+    public StrimziKafkaContainer withBrokerId(final int brokerId) {
+        this.brokerId = brokerId;
+        return self();
+    }
+
+    public StrimziKafkaContainer withKafkaVersion(final String kafkaVersion) {
+        this.kafkaVersion = kafkaVersion;
+        return self();
+    }
+
+    public StrimziKafkaContainer withStrimziTestContainerVersion(final String strimziTestContainerVersion) {
+        this.strimziTestContainerVersion = strimziTestContainerVersion;
+        return self();
+    }
+    public StrimziKafkaContainer withKraft(final boolean useKraft) {
+        this.useKraft = useKraft;
+        return self();
+    }
+
+    /**
+     * Fluent method, which sets @code{storageUUID}.
+     * The storage UUID must be the 16 bytes of a base64-encoded UUID.
+     *
+     * This method requires the broker to be created using Kraft.
+     *
+     * @param uuid the uuid, must not be {@code null} or blank.
+     * @return StrimziKafkaContainerBuilder instance
+     */
+    public StrimziKafkaContainer withStorageUUID(final String uuid) {
```

Comment:
What's the use case for allowing passing a uuid? How would be expect the caller to generate a string with the right syntax? Should we be using String? If so, should we validate it here?

Reply:
I am thinking about that if it's necessary to allow user to specify `storageUUID`...

Reply:
Did you come to a conclusion?

Reply:
@cescoffier implement it so I assume they are using it...Thus I think we should implement also some validation mechanism, which can be copy pasted from Kafka source:
```java
if (str.length() > 24) {
            throw new IllegalArgumentException("Input string with prefix `"
                + str.substring(0, 24) + "` is too long to be decoded as a base64 UUID");
        }

        ByteBuffer uuidBytes = ByteBuffer.wrap(Base64.getUrlDecoder().decode(str));
        if (uuidBytes.remaining() != 16) {
            throw new IllegalArgumentException("Input string `" + str + "` decoded as "
                + uuidBytes.remaining() + " bytes, which is not equal to the expected 16 bytes "
                + "of a base64-encoded UUID");
        }
```

Reply:
There is no real use case. At some point I would like to try to have the directory structure already created in the image. 

We can definitely remove that option and use the "default" value I provided (or any other default value)